tools/cli/adminCommands.go

```diff
@@ -503,6 +503,26 @@ func AdminCloseShard(c *cli.Context) {
 	}
 }
 
+// AdminDescribeShardDistribution describes shard distribution
+func AdminDescribeShardDistribution(c *cli.Context) {
+	adminClient := cFactory.ServerAdminClient(c)
+
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	req := &types.DescribeShardDistributionRequest{
+		PageSize: int32(c.Int(FlagPageSize)),
+		PageID:   int32(c.Int(FlagPageID)),
+	}
+
+	resp, err := adminClient.DescribeShardDistribution(ctx, req)
```

Comment:
nit: not sure if it's useful to have some logic there to automatically fetch next page upon user request. 

and  if `len(resp.Shards) < req.PageSize` it means there's no more pages?

Reply:
For the second one, maybe I can split NumberOfShards into 2 as TotalNumberOfShards and NumberOfShardsReturned; although this will require another idl change; just landed the previous one. Let me know if that sounds good or if you have any other ideas.

For the first one, we need to persist what user already saw to achieve this right? It might be better to give this control to the user. Or if there's any example, let me know.

Reply:
For the first one, what I am thinking is somewhat like our list workflow commands. The default pagesize is 10, and at the end of the each page, it will ask user if the next page should be displayed. If user hit enter then the next page will be displayed. It's kind of useful when user just want to get a sense of the shard distribution. I don't have strong opinion on it though. 

For the second one, I think the current impl is fine. I just want to confirm my understanding and see if this is something caller can rely on when trying to fetch all pages. 
Originally I was thinking we can follow the nextPageToken pattern we use for most APIs that needs pagination. But this impl also makes sense I think.